test/health_reporter_unittest.cc

```diff
@@ -0,0 +1,76 @@
+#include "../src/health_reporter.h"
+#include "gtest/gtest.h"
+#include <stdio.h>
+#include <sstream>
+
+namespace google {
+
+class HealthReporterUnittest : public ::testing::Test {
+ public:
+  HealthReporterUnittest() {}
+  ~HealthReporterUnittest() {
+    if(prefix.length() > 0) {
+      std::ostringstream oss;
+      oss << prefix << "_metadata_agent_unhealthy";
+      std::remove(oss.str().c_str());
+
+      oss.str("");
+      oss.clear();
+      oss << prefix << "_node_callback";
+      std::remove(oss.str().c_str());
+
+      oss.str("");
+      oss.clear();
+      oss << prefix << "_watch_pods";
+      std::remove(oss.str().c_str());
+    }
+  }
+ protected:
+  void SetUnhealthy(HealthReporter healthReporter, const std::string &state_name) {
+    healthReporter.SetUnhealthy(state_name);
+  }
+
+  bool ReportHealth(HealthReporter healthReporter) {
+    return healthReporter.ReportHealth();
+  }
+
+  bool GetTotalHealthState(HealthReporter healthReporter) {
+    return healthReporter.GetTotalHealthState();
+  }
+
+  std::string prefix;
```

Comment:
Please add a trailing `_` to instance variable names ([style guide](https://google.github.io/styleguide/cppguide.html#Variable_Names)).